storage/cloud-client/hmac_test.py

```diff
@@ -0,0 +1,88 @@
+# Copyright 2019 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Tests for hmac.py. Requires GOOGLE_CLOUD_PROJECT (valid project) and
+HMAC_KEY_TEST_SERVICE_ACCOUNT (valid service account email) env variables to be
+set in order to run.
+"""
+
+
+import os
+
+from google.cloud import storage
+import pytest
+
+import hmac
+
+
+PROJECT_ID = os.environ['GOOGLE_CLOUD_PROJECT']
+SERVICE_ACCOUNT_EMAIL = os.environ['HMAC_KEY_TEST_SERVICE_ACCOUNT']
+
+
+@pytest.fixture()
+def new_hmac_key():
+    """
+    Fixture to create a new HMAC key, and to guarantee all keys are deleted at
+    the end of each test.
+    """
+    hmac_key = hmac.create_key(PROJECT_ID, SERVICE_ACCOUNT_EMAIL)
```

Comment:
This fixture is inconsistently handling fixtures - sometimes it uses the snippet to perform the action and sometimes not. We should be consistent.

I think it makes more sense to avoid using the snippet, so that we don't have to worry about the logs from another snippet inadvertently showing up in the test or logs somewhere.

Reply:
I thought about this a bit-- the downside of not using the snippet is that then it's difficult to test the create_key snippet (because then I can't leverage the teardown in the fixture for the newly created key, so if assertions in the test fail the key will never be deleted). Any thoughts on how to get around this?

Reply:
Good points, thanks for the advice. I'll take your first suggestion.